pkg/git/git.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package git
+
+import (
+	"bytes"
+	"strings"
+
+	"go.uber.org/zap"
+	"os/exec"
+)
+
+
+func run(logger *zap.SugaredLogger, dir string, args ...string) (string, error) {
+	c := exec.Command("git", args...)
+	logger.Info(args)
+	var output bytes.Buffer
+	c.Stderr = &output
+	c.Stdout = &output
+	// This is the optional working directory. If not set, it defaults to the current
+	// working directory of the process.
+	if dir != "" {
+		c.Dir = dir
+	}
+	if err := c.Run(); err != nil {
+		outputList := strings.Split(output.String(), "fatal: ")
+		logger.Errorf("error %s", outputList[len(outputList)-1])
+		return outputList[len(outputList)-1], err
+	}
+	return "", nil
+}
```

Comment:
Me personally, I would opt for *not* using another program using command execution. This is generally speaking a dangerous pattern. If possible, I would recommend to always stick to what you have available with Go and libraries. With that in mind, maybe have a [look at how we solved the same issue in havener](https://github.com/homeport/havener/blob/cef60b9c6827b823173a4ecff8ecf83b0012b207/pkg/havener/locations.go#L67-L91) or another example. I can recommend https://github.com/go-git/go-git for this. We can do the pulling all in memory, which is another great advantage.

Reply:
I agree, we should try our best to stay away from additional software dependencies where possible.

Reply:
@HeavyWombat Thanks! I agree with you use `go-git` package rather than `git` command. I have switched to `go-git` package and use [`List`](https://github.com/go-git/go-git/blob/master/remote.go#L1027) func to do the same thing with `go ls-remote`.

But now, for a valid public repository url, if it doesn't exist, `List` func will return `authentication required` error. Such as `https://github.com/shipwright-io/build-fake`. If we put this error on build `Reason`, this will mislead users, so I convert this message to `repository not found`. 

Besides, an interesting thing is if url is invalid, then `List` func will return `repository not found`, such as a url named `foobar`. About this situation, I put `invalid source url` on build `Reason`.

Reply:
I´m resolving this, we now rely on a go pkg that seems to be the _de facto_ for git in go